### DISCUSSION TITLE
Testing with GPL code

### DISCUSSION THREAD

**Discussion by alemuller** [tree-sitter/tree-sitter#1478] [Q&A] [answered]
Does anyone know if I can *include in the public repository*, of a MIT licensed tree-sitter parser, tests with code snippets from manuals licensed under the different GPLs? How about Apache 2?

**Comment by Sjord** [comment, marked as answer]
IANAL. It seems to me the code snippets are copyrighted and cannot be used without permission from the license. For GPL that means that your entire project must also be licensed under GPL. For Apache that means that that at least the test snippets must also be licensed under Apache.

Perhaps the easiest way out is to ask the project maintainer. If they own the copyright they can give you permission outside of the license.

**Reply by theHamsta** [reply]
You could have a separate project that has the GPL tests and your MIT project as a dependency.

As far as I know doesn't GPL force the license on code that is not derived from GPL works. Just GPL terms become effective on distributions that contain GPL portions or link against GPL projects.
